doc/docs/Python_Tutorials/Mode_Decomposition.md

```diff
@@ -922,17 +922,26 @@ plt.show()
-           mp.Source(mp.GaussianSource(fcen,fwidth=0.05*fcen), component=mp.Ey, center=src_pt, size=mp.Vector3(0,sy,0), amplitude=1j)]
+sources = [mp.Source(mp.GaussianSource(fcen,fwidth=0.05*fcen),
+                     component=mp.Ez,
+                     center=src_pt,
+                     size=mp.Vector3(0,sy,0)),
+           mp.Source(mp.GaussianSource(fcen,fwidth=0.05*fcen),
+                     component=mp.Ey,
+                     center=src_pt,
+                     size=mp.Vector3(0,sy,0),
+                     amplitude=1j)]
 ```
 
-Note that even though the $J_y$ current amplitude is complex in this example, only its real part is used and the resulting fields are therefore still real (the default).
+Note: when imparting a phase offset using a complex `amplitude`, it is *not* necessary to set `force_complex_fields=True` in the `Simulation` constructor since the real part of the current *includes* the phase offset.
 
-The figure below shows a snapshot of $E_z$ within the cell for four different cases: phase delays ($\Delta nd/\lambda$) of 0.5 and 1.0, and planewave circular polarization of $E_z + iE_y$ and $E_z - iE_y$. The empty regions on the cell sides are PMLs. The thin solid black line denotes the boundary between the grating (on the left) and air. As expected, for $\Delta nd/\lambda=0.5$ there is just a single ±1 diffraction order which depends on the chirality of the input planewave (this is not the case for a linear-polarized planewave). The angle of this diffracted order (±4.8°) agrees with the analytic result. Snapshots of $E_y$ are similar.
+The figure below shows a snapshot of $E_z$ for four different cases: phase delays ($\Delta nd/\lambda$) of 0.5 and 1.0, and circular-polarized planewave sources of $E_z + iE_y$ and $E_z - iE_y$. The empty regions on the cell sides are PMLs. The thin solid black line denotes the boundary between the grating (on the left) and air. As expected, for $\Delta nd/\lambda=0.5$ there is just a single ±1 diffraction order which depends on the chirality of the input planewave (this is not the case for a linear-polarized planewave). The angle of this diffracted order (±4.8°) agrees with the analytic result. Snapshots of $E_y$ are similar.
 
 <center>
 ![](../images/polarization_grating_diffraction_orders.png)
 </center>
+
+For computing the mode coefficient of an [elliptical-polarized](https://en.wikipedia.org/wiki/Elliptical_polarization) mode in 3d, the mode coefficients of two orthogonal linear-polarized modes ($\mathcal{S}$ and $\mathcal{P}$ polarizations) should first be computed separately using the [`DiffractedPlanewave`](../Python_User_Interface.md#diffractedplanewave) object which is passed as the `bands` parameter to `get_eigenmode_coefficients`. These complex mode coefficients for the two polarizations can then be combined in post processing via a linear combination to compute the mode coefficient for any arbitrary elliptical-polarized mode.
```

Comment:
You could also specify an elliptically polarized planewave directly by specifying both s and p for a single component.